### PR TITLE
vuejs 2-crud-with-localstorage

### LINKED ISSUE #1220: Creating a Vuejs2 CRUD application with local storage
### Brief Summary: 

A Vuejs2 CRUD application with local storage. Local storage helps persist data so users don't lose their state of the app between sessions. It stores and retrieves data in Web Storage API in modern browsers, and can be used completely offline as a data storage.

### Key Takeaways:

By the end of the article the student should be able to:
1. Create a Vuejs project with Vue CLI
2. Install the Vuejs dev tools browser extension.
3. Creating the Books Component and BookItem component.
4. Work with data.
5. Save and load data from Local Storage

### References: 
N/A

### PR DESCRIPTION
**Pull request by EspiraMarvin**
this fixes #1220

### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by EspiraMarvin** [comment]
@geoffrey45 done

**Comment by geoffrey45** [comment]
@EspiraMarvin You have an author folder in your article directory. Please delete this folder ans commit again.

**Comment by EspiraMarvin** [comment]
Link to my issue #1220

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/vue2-crud-app-with-localstorage/index.md [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+        // clear input field
+        this.title = '';
+        this.edit = false;
+    }
+}
+``` 
+
+By now you can click on the edit button and the input field will be populated with the book title. 
+<<<<<<< HEAD
+The Watch method comes in handy again, to help us watch for any changes in the editBook data, we set ```deep:true``` property to let the Vue instance continuously watch for changes, so while editing the edit property will always be true, it also watches the title property and if it’s empty it sets the edit property to false, here we don’t need the deep property.
```
Comment (geoffrey45) [inline articles/vue2-crud-app-with-localstorage/index.md:452]: This sentence is too long. Consider breaking it into simpler sentences.

File: articles/vue2-crud-app-with-localstorage/index.md [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+        handler() {
+            if (this.title === ''){
+                this.edit = false;
+            }
+        }
+    }
+}
+```
+
+Go back to the App.vue file, after editing a title an event ```edit-book-event``` is sent to App.vue file we assign the event to a method in order to save the changes to localStorage. Update the code
```
Comment (geoffrey45) [inline articles/vue2-crud-app-with-localstorage/index.md:476]: This sentence is too long. Consider breaking it into simpler sentences.

File: articles/vue2-crud-app-with-localstorage/index.md [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+  - url: /engineering-education/vue2-crud-app-with-localstorage/hero.jpg
+    alt: vuejs 2 crud app with localstorage
+---
+![local storage with vuejs](/engineering-education/vue2-crud-app-with-localstorage/hero.jpg)
+
+
+### Introduction
+
+Vuejs is a progressive, approachable, and versatile front-end JavaScript framework that’s highly being adopted by developers due to its simplicity, short learning curve, and its little bundle size.
+
```
Comment (geoffrey45) [inline articles/vue2-crud-app-with-localstorage/index.md:21]: This sentence is too long. Consider breaking it into simpler sentences.

File: articles/vue2-crud-app-with-localstorage/index.md [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+---
+ 
+By now you should have already come across the acronym CRUD plenty of times, these are create, read, update, and delete functions required to implement persistent storage application.
+
```
Comment (geoffrey45) [inline articles/vue2-crud-app-with-localstorage/index.md:25]: This sentence is too long. Consider breaking it into simpler sentences.

File: articles/vue2-crud-app-with-localstorage/index.md [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+                }
+                this.title = ''
+            }
+        }
+
+    }
+</script>
+```
+This code has a form that you can use to add a book, it has a method addBook, and a vue-directive v-model which creates a 2-way binding between user-input and vuejs component, any changes to an input value the bound data will be changed and vice versa. In this case the title.
+
```
Comment (geoffrey45) [inline articles/vue2-crud-app-with-localstorage/index.md:236]: This sentence is too long. Consider breaking it into simpler sentences.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@EspiraMarvin Great article. Teaches us how to use local storage in Vuejs in a simple and straightforward manner. Just a few issues. Please fix these issues and ping me when you are done.

File: articles/vue2-crud-app-with-localstorage/index.md [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+<AddBookItem v-model="editBook.title" v-on:add-book-event="addBookItem"  v-bind:editBook="editBook" v-on:edit-book-event="editBookItemEvent" />
+```
+Now we create a method `editBookItemEvent()` to handle saving of data. In the method we find the index of the ID’s object. This index will be used to reassign the title of the book being edited. If you’ve reached this far you can edit a book title.  Add the code below to `methods` in `App.vue`.
+```javascript
+editTodoItemEvent(todoItem){
+    //find index of this id's object
+    var objIndex = this.todo_items.findIndex(obj => obj.id === todoItem.id);
+    //update the item
+    this.todo_items[objIndex].title = todoItem.title;
+}
```
Comment (geoffrey45) [inline articles/vue2-crud-app-with-localstorage/index.md:436]: Looks like you are using the wrong block of code here. Please replace this with the correct books code.

File: articles/vue2-crud-app-with-localstorage/index.md [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+              <i class="glyphicon glyphicon-trash" @click="$emit('del-book-item', book.id)">delete</i>
+            </button>
+        </span>
+    </div>
+</template>
+```
+An event called `edit-book-item` is emitted and passed with it the book id to its parent(`Books.vue`). In `Books.vue` listen to the event and assign it to a method called `editBookMethod()` as shown below.
+```html
+<BookItem  v-bind:book="book"  v-on:del-book-item="delBookMethod" v-on:edit-todo-item="editBookMethod" />
+```
```
Comment (geoffrey45) [inline articles/vue2-crud-app-with-localstorage/index.md:330]: Please replace `v-on:edit-todo-item` with the correct piece of code

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@EspiraMarvin almost through with the tech review. Please fix the minor issues below. Please let me know when you are through

**Comment by hectorkambow** [comment]
As an FYI we'll be adding a tiered payout label due to the level of revisions and reviews.

**Comment by EspiraMarvin** [comment]
@hectorkambow I was inquiring is this not a tech tutorial?

**Pull request merged**